### ISSUE TITLE
Question: How should you read the logged images?

### ISSUE THREAD

**Issue by TaleirOfDeynai** [rinongal/textual_inversion#99]
So, reading through other issues, I understand that any `loss` statistic shown is pretty much useless to determine how close training is to some meaning of "optimal".  So, I assume you're meant to only use the logged images from `images/train` for this instead.

But, I'm having a hard time figuring out exactly what the images are telling me; **this has been my current assumption**:

- `inputs` and `reconstruction` are canaries to detect training going sideways, like a way-too-high learning rate or other misconfiguration.  In other words, if these look *very* messed up or too distorted, something is very wrong.
- `conditioning` is the prompt in use to generate the latter images.
- `samples` is the prompt with balanced weighting to try and track how well the concept being trained is generalizing across domains.
- `samples_scaled` is the prompt with the embedding given very heavy weight so you can track exactly what aspects of the concept are being isolated.  That is, these are the concept depicted in a very pure form.

I've been using `samples_scaled` to track the overall health of the training, like "is it actually picking up on the concept?"  And I've been waiting for images from `samples` to reach a decent level of coherency to tell me when the embedding "fits" how it is used in the conditioning prompts.

**So, my question is:** did I interpret this right?  And if not, what is the correct way to read them?

Also it might be helpful to add something to the readme clarifying how to gauge the progress of training so no one has to bug you here with inane questions. xP

**Comment by badcode6** [comment]
I have the same question. why are samples and samples_scaled pretty much total opposites? samples are pretty much just shapes, while samples_scaled resemble the input more and more (on colab anyway)

**Comment by TaleirOfDeynai** [comment]
As training progresses, I see more and more hints of the trained concept come through in the `samples`, but it takes a while with a larger training set.

This especially comes through if you train for a bit, stop, then resume training from the last checkpoint.  Because the same seed is used, it will use the same latent noise as the original run when generating the sample images; you can sort of see how the training has adjusted its interpretation of that same latent noise.  You can flip between the previous version and the new version and see similarities in the structures.

Kinda makes me wonder if I should always be using a new seed when restarting training though...  :thinking:

**Comment by TaleirOfDeynai** [comment]
Got it.  Thanks for the answer~  :grin:

I guess I'll close this, but I'd still suggest adding this explanation to the readme where it's easier to find.  It would help us bumbling newcomers who are reluctant to immediately jump into one of the related Discord servers out a lot.  Heh...
